synapse/storage/engines/postgres.py

```diff
@@ -45,6 +45,15 @@ def _disable_bytes_adapter(_: bytes) -> NoReturn:
 
         psycopg2.extensions.register_adapter(bytes, _disable_bytes_adapter)
         self.synchronous_commit: bool = database_config.get("synchronous_commit", True)
+        # Set the statement timeout to 1 hour by default.
```

Comment:
Will the cancelled queries show up anywhere? I guess if the 1 hour is > your slow query log config it'll show there?

Reply:
yeah, this will go to the slow query log as per usual, plus you'll get an exception in Python which will most likely generate a traceback

Reply:
After a quick test I am getting:
```
ERROR:  canceling statement due to statement timeout
STATEMENT:  SELECT pg_sleep(10);
```